staging/src/k8s.io/kubectl/pkg/cmd/util/helpers.go

```diff
@@ -625,8 +634,38 @@ func DumpReaderToFile(reader io.Reader, filename string) error {
 	return nil
 }
 
-func GetServerSideApplyFlag(cmd *cobra.Command) bool {
-	return GetFlagBool(cmd, "server-side")
+type ServerSideApplyStrategy string
+
+const (
+	ServerSide ServerSideApplyStrategy = "true"
+
+	ClientSide ServerSideApplyStrategy = "false"
+
+	ServerSideAuto ServerSideApplyStrategy = "auto"
+)
+
+func GetServerSideApplyStrategy(cmd *cobra.Command) (ServerSideApplyStrategy, error) {
+	var serverSideFlag = GetFlagString(cmd, "server-side")
+	invalidOpt := fmt.Errorf(`invalid server-side value (%v) - must be one of true or false`, serverSideFlag)
+	if AutoServerSide.IsEnabled() {
+		invalidOpt = fmt.Errorf(`invalid server-side value (%v) - must be one of true, false or auto`, serverSideFlag)
+	}
+
+	b, err := strconv.ParseBool(serverSideFlag)
+	// The flag is not a boolean - we only accept auto
+	if err != nil {
+		if AutoServerSide.IsEnabled() && serverSideFlag == string(ServerSideAuto) {
+			return ServerSideAuto, nil
+		}
+		return ClientSide, invalidOpt
+	}
+	if b {
+		return ServerSide, nil
+	} else if !b {
+		return ClientSide, nil
+	}
+
+	return ClientSide, invalidOpt
```

Comment:
I think this line is unreachable?  And if that's true then I don't think you need to preconstruct invalidOpt, it is never reused.

Reply:
I think this would be the case when a non bool that is not "" or "auto"